### DISCUSSION TITLE
Can not access static files with express

### DISCUSSION THREAD

**Discussion by vinsdesc** [vercel/community#1663] [Help] [answered]
### Summary

After following this guide i can't get access to my static files on vercel.

although the build works
<html>
<body>

Generated build outputs:
--
16:08:54.933 | - Static files: 41
16:08:54.933 | - Prerenders: 0
16:08:54.933 | - Serverless Functions: 1
16:08:54.933 | - Edge Functions: 0
16:08:54.933 | Serverless regions: Washington, D.C., USA
16:08:54.933 | Deployed outputs in 2s
16:08:55.243 | Build completed. Populating build cache...
16:08:57.060 | Uploading build cache [4.18 MB]...

</body>
</html>

I can even see it the output source section 

![image](https://user-images.githubusercontent.com/2938854/222184180-2f2d523d-9e45-4097-a77d-a64433f79eec.png)

You can test it here https://bsr-api.vercel.app/

**Comment by vinsdesc** [comment]
Here is some content to reproduce it.

### **file structure**
![image](https://user-images.githubusercontent.com/2938854/222186964-f9dd0636-10ba-49b9-9dfa-3e371c7b3173.png)

### index.js
```
require('dotenv').config();
const express = require('express');
const app = express();
const port = 4000;


app.use(express.static('public'));

app.get('/', (req, res) => res.send('<h1 style="color:blue;">Test</h1> <img src="/images/branche.svg" />'));


app.listen(port, () => {
  console.log(`BSR app listening on port http://localhost:${port}`);
});

module.exports = app
```

### vercel.json
```
{
  "builds": [
      {
          "src": "index.js",
          "use": "@vercel/node"
      },
      {
        "src": "public/**",
        "use": "@vercel/static"
    }
  ],
  "routes": [
      {
        "src": "/(.*)",
        "dest": "index.js"
      },

    {
        "src": "/",
        "dest": "public/test.html"
    },
    {
        "src": "/(.+)",
        "dest": "public/$1"
    }
  ]
}
```

**Comment by amyegan** [comment, marked as answer]
Hi @vinsdesc! Did you follow the guide on [using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel)? Or was it a different guide?

**Reply by vinsdesc** [reply]
Yes this one indeed

**Reply by amyegan** [reply]
It looks like your file structure doesn't quite match that guide and your current `vercel.json` is causing some irregular routing behavior which blocked your image. I put together a quick example repo (https://github.com/amyegan/express-1663) combining the code you shared and the recommended `/api` directory approach from that guide. The main differences of note are that `index.js` is now in the `/api` directory and the `vercel.json` config is different. You can see it in action at https://express-1663.vercel.app
